src/binary_path.hpp

```diff
@@ -0,0 +1,100 @@
+/*
+   Copyright (C) 2020 by the Battle for Wesnoth Project https://www.wesnoth.org/
+
+   This program is free software; you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation; either version 2 of the License, or
+   (at your option) any later version.
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY.
+
+   See the COPYING file for more details.
+*/
+
+/**
+ * @file
+ * Declarations for File-IO.
+ */
+
+#pragma once
+
+#include <algorithm>
+#include <ctime>
+#include <iosfwd>
+
+#include <memory>
+#include <string>
+#include <vector>
+
+#include "wml_path.hpp"
+
+#include "exceptions.hpp"
+#include "serialization/string_utils.hpp"
+
+class config;
+class game_config_view;
+
+namespace filesystem {
+
+using scoped_istream = std::unique_ptr<std::istream>;
+using scoped_ostream = std::unique_ptr<std::ostream>;
+
+
+/**
+ *  The paths manager is responsible for recording the various paths
+ *  that binary files may be located at.
+ *  It should be passed a config object which holds binary path information.
+ *  This is in the format
+ *@verbatim
+ *    [binary_path]
+ *      path=<path>
+ *    [/binary_path]
+ *  Binaries will be searched for in [wesnoth-path]/data/<path>/images/
```

Comment:
Same issue as the previous comment. I realize this was copy-pasted from elsewhere but you may as well fix it at the same time, right?